packages/pwa-buildpack/lib/Utilities/addImgOptMiddleware.js

```diff
@@ -29,9 +37,20 @@ function addImgOptMiddleware(app, config) {
     let cacheMiddleware;
     let imageopto;
     try {
-        cacheMiddleware = cache(cacheExpires, hastily.hasSupportedExtension, {
-            debug: cacheDebug
-        });
+        cacheMiddleware = cache(
+            cacheExpires,
+            (req, res) =>
+                hastily.hasSupportedExtension(req) && res.statusCode === 200,
+            {
+                debug: cacheDebug,
+                appendKey: req =>
+                    IsWebpSupported(req) ? 'supported' : 'regular',
```

Comment:
This definitely solves the problem, but I'm concerned why it is needed. The `Vary` header in the response contains the `accept` header, and this lib doesn't seem to be respecting it. If we proceed with this workaround, I would probably just want `appendKey: req => req.headers.accept` while we investigate why this lib isn't respecting the `Vary` header.